var/spack/repos/builtin/packages/intel-oneapi/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import tempfile
+import os
+import glob
+
+
+class IntelOneapi(Package):
+    """
+    Includes the icx/ifx compiler executables.
+    """
+
+    homepage = "https://software.intel.com/content/www/us/en/develop/tools/oneapi.html"
+
+    version('2021.1.9.2205', sha256='8926a3001e61edbb293cb607beb8eb3a3511330a4625c8f3b1b51603426f121a', url='https://registrationcenter-download.intel.com/akdlm/irc_nas/16949/l_HPCKit_b_2021.1.9.2205_offline.sh', expand=False)
+
+    phases = ['install']
+
+    def install(self, spec, prefix):
+        bash = Executable('bash')
+
+        # Capture logs written in /tmp
+        tmpdir = tempfile.mkdtemp(prefix='spack-intel-')
+        bash.add_default_env('TMPDIR', tmpdir)
+
+        # Need to set HOME to avoid using ~/intel
+        bash.add_default_env('HOME', prefix)
+
+        bash('./l_HPCKit_b_%s_offline.sh' %
+             spec.versions.lowest(),
+             '-s', '-a', '-s', '--action', 'install',
+             '--eula', 'accept',
+             '--components',
+             ('intel.oneapi.lin.dpcpp-cpp-compiler-pro'
+              ':intel.oneapi.lin.ifort-compiler'),
+             '--install-dir', prefix)
+
+        # preserve config and logs
```

Comment:
I am checking if there is a command line option that would make this more direct.

Reply:
Installer team confirmed that setting TMPDIR and HOME is the best way to control where the files are put.

Reply:
OK - based on that it sounds like this is the best that can be done?

I'll say I originally copied this over from https://github.com/spack/spack/pull/19033 without giving it much thought, but we typically reserver `.spack/` for internal Spack usage. In this case I think it's worthwhile to keep logs. Also note that we generally capture everything printed to `stdout`/`stderr` and store *that* in `.spack/spack-build-out.txt`, so unless these logs are created by opening/writing files this would already be covered by that.

Finally, I'm inspecting the `.spack` directory and not seeing any additional log files there (i.e. anything that fits the pattern passed to `glob`). Is it possible for no such files to exist?

Reply:
The installer puts a log file in the install dir that summarizes the entire install. I think that is sufficient. In tmpdir it writes detailed log files for the install of each component. I don't think you need it, but if you want it, then this glob would find the files relative to tmpdir: */*/*/*.log. Note that this could change in the future because they assume a human is looking at the logs.